### PR TITLE
Add support for ACPI IVRS table for HSI security attributes

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [x] New plugin

### CODE REVIEW

File: docs/hsi.md [hunk 1]
```diff
@@ -436,15 +436,16 @@ To meet HSI-2 on systems that run this test, the result must be `valid`. *[v1.5.
 
 - [Linux Kernel TPM Documentation](https://www.kernel.org/doc/html/latest/security/tpm/tpm_event_log.html)
 
-<a id="org.fwupd.hsi.AcpiDmar"></a>
+<a id="org.fwupd.hsi.PrebootDma"></a>
```
Comment (hughsie) [inline docs/hsi.md:439]: I think that we need to have both, i.e. two <a> sections with the old name too.

File: docs/hsi.md [hunk 2]
```diff
@@ -454,11 +455,12 @@ This would allow subverting the Secure Boot protection, and also invalidate any
 - `not-valid`: could not determine state
 - `not-enabled`: was not enabled
 
-To meet HSI-3 on systems that run this test, the result must be `enabled`. *[v1.5.0]*
+To meet HSI-3 on systems that run this test, the result must be `enabled`. *[v1.8.0]*
```
Comment (hughsie) [inline docs/hsi.md:458]: I think we need to capture that the attribute existed in 1.5.0 for Intel and 1.8.0 for AMD.

File: libfwupd/fwupd-security-attr-private.h [hunk 3]
```diff
@@ -14,13 +14,13 @@
 G_BEGIN_DECLS
 
 /**
- * FWUPD_SECURITY_ATTR_ID_ACPI_DMAR:
+ * FWUPD_SECURITY_ATTR_ID_PREBOOT_DMA_PROTECTION:
  *
  * Host Security ID attribute for ACPI DMAR table
  *
```
Comment (hughsie) [inline libfwupd/fwupd-security-attr-private.h:20]: The old name for this was org.fwupd.hsi.AcpiDmar etc.

Comment (superm1) [reply libfwupd/fwupd-security-attr-private.h:20]: fixed

File: docs/hsi.md (hunk 2 above)
Comment (superm1) [reply docs/hsi.md:459]: fixed

File: docs/hsi.md (hunk 1 above)
Comment (superm1) [reply docs/hsi.md:440]: fixed

**Review by hughsie — APPROVED** [review]

**Pull request merged**